packages/flutter/lib/src/material/typography.dart

```diff
@@ -6,623 +6,449 @@ import 'package:flutter/foundation.dart';
+/// [ScriptCategory.dense] scripts - for text styles that are smaller than the
+/// title style - are one unit larger than they are for
+/// [ScriptCategory.englishLike] scripts.
+enum ScriptCategory {
+  /// The languages of Western, Central, and Eastern Europe and much of
+  /// Africa are typically written in the Latin alphabet. Vietnamese is a
+  /// notable exception in that, while it uses a localized form of the Latin
+  /// writing system, its accented glyphs can be much taller than those
+  /// found in Western European languages. The Greek and Cyrillic writing
+  /// systems are very similar to Latin.
+  englishLike,
+
+  /// Language scripts that require extra line height to accommodate larger
+  /// glyphs, including Chinese, Japanese, and Korean.
+  dense,
+
+  /// Language scripts that require extra line height to accommodate
+  /// larger glyphs, including South and Southeast Asian and
+  /// Middle-Eastern languages, like Arabic, Hindi, Telugu, Thai, and
+  /// Vietnamese.
+  tall,
+}
 
-
-/// Material design text theme.
+/// The color and geometry [TextThemes] for Material apps.
+///
+/// The text themes provided by the overall [Theme], like
+/// [ThemeData.textTheme], are based on the current locale's
+/// [MaterialLocalizations.scriptCategory] and are created
+/// by merging a color text theme, [black] or [white]
+/// and a geometry text theme, one of [englishLike], [dense],
+/// or [tall], depending on the locale.
 ///
-/// Definitions for the various typographical styles found in material design
-/// (e.g., headline, caption). Rather than creating a [TextTheme] directly,
-/// you can obtain an instance as [Typography.black] or [Typography.white].
+/// To lookup a localized text theme use
+/// `Theme.of(context).textTheme` or
+/// `Theme.of(context).primaryTextTheme` or
+/// `Theme.of(context).accentTextTheme`.
 ///
-/// To obtain the current text theme, call [Theme.of] with the current
-/// [BuildContext] and read the [ThemeData.textTheme] property.
+/// The color text themes are [blackMountainView],
+/// [whiteMountainView], and [blackCupertino] and [whiteCupertino]. The
+/// MountainView theme [TextStyles] are based on the Roboto fonts and the
+/// Cupertino themes are based on the San Francisco fonts.
 ///
-/// The following image [from the material design
-/// specification](https://material.io/go/design-typography#typography-styles)
-/// shows the recommended styles for each of the properties of a [TextTheme].
-/// This image uses the `Roboto` font, which is the font used on Android. On
-/// iOS, the [San Francisco
-/// font](https://developer.apple.com/ios/human-interface-guidelines/visual-design/typography/)
-/// is automatically used instead.
+/// Two sets of geometry themes are provided: 2014 and 2018. The 2014 themes
+/// correspond to the original version of the Material Design spec and are
+/// the defaults. The 2018 themes correspond the current spec and feature
```

Comment:
s/correspond the current spec/correspond to the second iteration of the specification/

Reply:
Done